### PR TITLE
Case-insensitive / search-style Entry property fetching

### PR DESCRIPTION
**Pull request by benjspriggs**
## summary
This PR implements RegExp Property searching for Entries (described in #213), and updates unit tests for `Entry` accordingly`.

### CODE REVIEW

**Comment by perry-mitchell** [comment]
Hi!

Thanks for this. Before I review, could you please remove the changes you made for the .md files and the archive test file? These are generated upon release only.

**Comment by perry-mitchell** [comment]
Thanks @benjspriggs.. The tests are failing due to possibly some dependency version change.. Probably with dropbox. We'll need to get that fixed before merging.

**Comment by perry-mitchell** [comment]
This may be fixed by #240, will wait for tests to pass.

**Comment by perry-mitchell** [comment]
Hi @benjspriggs - Could you please rebase/merge upstream? I've made changes to master that should fix the tests. There may be conflicts with some files that were auto-formatted for you, but they should be the same.

**Comment by benjspriggs** [comment]
@perry-mitchell I've rebased with some of the changes, should be ready for comment now.

**Comment by perry-mitchell** [comment]
Looks good @benjspriggs - Thanks for that. I'll merge.

File: source/node/Entry.js [hunk 1]
```diff
@@ -212,6 +212,31 @@ class Entry {
...
+     * @returns {Array} The values of the properties that matched the expression.
+     * @memberof Entry
+     */
+    getProperties(propertyExpression) {
+        if (typeof propertyExpression === "undefined") {
+            return this.getProperty();
+        }
+
+        const raw = this._getRemoteObject().properties;
+        var result = [];
```
Comment (sallar) [inline source/node/Entry.js]: Im a bit late but shouldnt this be removed?

Comment (benjspriggs) [reply source/node/Entry.js]: It should, I opened a new PR to remove it (#241).

**Comment by perry-mitchell** [comment]
It’s ok I can tidy up if need be before releasing. Cheers!

**Comment by perry-mitchell** [comment]
Just FYI I've updated the function to output an object instead of an array as it's more useful. `Object.values()` can be used to still get the original return structure if need be.

**Pull request merged**